# SURPLUS SHARE TREATY

Surplus share treaties are similar to quota share treaties, but the primary insurer cedes a fixed
percentage of the amount of insurance that exceeds a specified retention level. 

This allows the insurer to retain a portion of the risk while passing on excess risk to the reinsurer.

We upload the required packages needed for the analysis:

In [1]:
import numpy as np
import scipy.stats as stats


Simulation of our data takes place in respect to the time value of money, where the claims and the premiums are ascertained.

In [2]:
def simulate_policy_data(policy_count, average_policy_premium, average_claim_loss):
    np.random.seed(0)
    policy_premiums = np.random.exponential(scale=average_policy_premium, size=policy_count)
    claim_losses = np.random.exponential(scale=average_claim_loss, size=policy_count)
    return policy_premiums, claim_losses

Surplus share is ascertained on policy level

In [4]:
def calculate_surplus_share_retention(policy_premiums, surplus_share_percentage):
    return np.percentile(policy_premiums, 100 * (1 - surplus_share_percentage))

The excessive loss from policies is also attained, given the policy structure and claim losses

In [5]:
def identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention):
    return claim_losses[policy_premiums > surplus_share_retention]

The Reinsurer liability is to attained with respect to the surplus share percentage 

In [6]:
def calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage):
    return np.sum(excess_loss_policies) * surplus_share_percentage

Value At Risk and Confidence Tail Expectation is to be defined for calculation of the needed files

In [7]:
def calculate_var_and_cte(excess_loss_policies, confidence_level):
    excess_loss_sorted = np.sort(excess_loss_policies)
    var_index = int(len(excess_loss_sorted) * (1 - confidence_level))
    var = excess_loss_sorted[var_index]
    cte = np.mean(excess_loss_sorted[:var_index])
    return var, cte


Data is defined as per the  data set that is needed per confidence level and surplus share level which is defined.

In [8]:
# Input data
policy_count = 1000
average_policy_premium = 10000
average_claim_loss = 5000
surplus_share_percentage = 0.2
confidence_level = 0.95

Next we simulate the policy data 

In [9]:
# Simulate policy data
policy_premiums, claim_losses = simulate_policy_data(policy_count, average_policy_premium, average_claim_loss)

Surplus Share retention is calculated with reference to the share percentage 

In [10]:
# Calculate surplus share retention
surplus_share_retention = calculate_surplus_share_retention(policy_premiums, surplus_share_percentage)

The excess loss policies are to be identified in order to attain the quantity of policies that are loss making and exceed the retention 

In [11]:
# Identify excess loss policies
excess_loss_policies = identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention)

The reinsurer liability is what is assesed next, to attain the financial position of the reinsurer.

In [12]:
# Calculate reinsurer liability
reinsurer_liability = calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage)

The VaR and CTE is to be ascertained and defined with reference to the confidence level

In [13]:
# Calculate Value at Risk (VaR) and Conditional Tail Expectation (CTE)
var, cte = calculate_var_and_cte(excess_loss_policies, confidence_level)

Lastly the results are displayed

In [14]:
# Print results
print(f"Surplus Share Retention: ${surplus_share_retention:,.2f}")
print(f"Reinsurer Liability: ${reinsurer_liability:,.2f}")
print(f"Value at Risk (VaR) at {confidence_level * 100}%: ${var:,.2f}")
print(f"Conditional Tail Expectation (CTE) at {confidence_level * 100}%: ${cte:,.2f}")

Surplus Share Retention: $16,420.62
Reinsurer Liability: $214,255.35
Value at Risk (VaR) at 95.0%: $217.09
Conditional Tail Expectation (CTE) at 95.0%: $110.87


# 85% CONFIDENCE INTERVAL

In [15]:
def simulate_policy_data(policy_count, average_policy_premium, average_claim_loss):
    np.random.seed(0)
    policy_premiums = np.random.exponential(scale=average_policy_premium, size=policy_count)
    claim_losses = np.random.exponential(scale=average_claim_loss, size=policy_count)
    return policy_premiums, claim_losses

In [16]:
def calculate_surplus_share_retention(policy_premiums, surplus_share_percentage):
    return np.percentile(policy_premiums, 100 * (1 - surplus_share_percentage))

In [17]:
def identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention):
    return claim_losses[policy_premiums > surplus_share_retention]

In [18]:
def calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage):
    return np.sum(excess_loss_policies) * surplus_share_percentage

In [19]:
def calculate_var_and_cte(excess_loss_policies, confidence_level):
    excess_loss_sorted = np.sort(excess_loss_policies)
    var_index = int(len(excess_loss_sorted) * (1 - confidence_level))
    var = excess_loss_sorted[var_index]
    cte = np.mean(excess_loss_sorted[:var_index])
    return var, cte

In [20]:
# Input data
policy_count = 1000
average_policy_premium = 10000
average_claim_loss = 5000
surplus_share_percentage = 0.2
confidence_level = 0.85

In [21]:
# Simulate policy data
policy_premiums, claim_losses = simulate_policy_data(policy_count, average_policy_premium, average_claim_loss)

In [22]:
# Calculate surplus share retention
surplus_share_retention = calculate_surplus_share_retention(policy_premiums, surplus_share_percentage)

In [23]:
# Identify excess loss policies
excess_loss_policies = identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention)

In [24]:
# Calculate reinsurer liability
reinsurer_liability = calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage)

In [25]:
# Calculate Value at Risk (VaR) and Conditional Tail Expectation (CTE)
var, cte = calculate_var_and_cte(excess_loss_policies, confidence_level)

In [26]:
# Print results
print(f"Surplus Share Retention: ${surplus_share_retention:,.2f}")
print(f"Reinsurer Liability: ${reinsurer_liability:,.2f}")
print(f"Value at Risk (VaR) at {confidence_level * 100}%: ${var:,.2f}")
print(f"Conditional Tail Expectation (CTE) at {confidence_level * 100}%: ${cte:,.2f}")

Surplus Share Retention: $16,420.62
Reinsurer Liability: $214,255.35
Value at Risk (VaR) at 85.0%: $792.54
Conditional Tail Expectation (CTE) at 85.0%: $376.31


# 75% CONFIDENCE INTERVAL

In [27]:
def simulate_policy_data(policy_count, average_policy_premium, average_claim_loss):
    np.random.seed(0)
    policy_premiums = np.random.exponential(scale=average_policy_premium, size=policy_count)
    claim_losses = np.random.exponential(scale=average_claim_loss, size=policy_count)
    return policy_premiums, claim_losses

In [28]:
def calculate_surplus_share_retention(policy_premiums, surplus_share_percentage):
    return np.percentile(policy_premiums, 100 * (1 - surplus_share_percentage))

In [29]:
def identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention):
    return claim_losses[policy_premiums > surplus_share_retention]

In [30]:
def calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage):
    return np.sum(excess_loss_policies) * surplus_share_percentage

In [31]:
def calculate_var_and_cte(excess_loss_policies, confidence_level):
    excess_loss_sorted = np.sort(excess_loss_policies)
    var_index = int(len(excess_loss_sorted) * (1 - confidence_level))
    var = excess_loss_sorted[var_index]
    cte = np.mean(excess_loss_sorted[:var_index])
    return var, cte

In [32]:
# Input data
policy_count = 1000
average_policy_premium = 10000
average_claim_loss = 5000
surplus_share_percentage = 0.2
confidence_level = 0.75

In [33]:
# Simulate policy data
policy_premiums, claim_losses = simulate_policy_data(policy_count, average_policy_premium, average_claim_loss)

In [34]:
# Calculate surplus share retention
surplus_share_retention = calculate_surplus_share_retention(policy_premiums, surplus_share_percentage)

In [35]:
# Identify excess loss policies
excess_loss_policies = identify_excess_loss_policies(policy_premiums, claim_losses, surplus_share_retention)

In [36]:
# Calculate reinsurer liability
reinsurer_liability = calculate_reinsurer_liability(excess_loss_policies, surplus_share_percentage)

In [37]:
# Calculate Value at Risk (VaR) and Conditional Tail Expectation (CTE)
var, cte = calculate_var_and_cte(excess_loss_policies, confidence_level)

In [38]:
# Print results
print(f"Surplus Share Retention: ${surplus_share_retention:,.2f}")
print(f"Reinsurer Liability: ${reinsurer_liability:,.2f}")
print(f"Value at Risk (VaR) at {confidence_level * 100}%: ${var:,.2f}")
print(f"Conditional Tail Expectation (CTE) at {confidence_level * 100}%: ${cte:,.2f}")

Surplus Share Retention: $16,420.62
Reinsurer Liability: $214,255.35
Value at Risk (VaR) at 75.0%: $1,596.54
Conditional Tail Expectation (CTE) at 75.0%: $694.48


# ANALYSIS

# At 95% we attained the below output:

Surplus Share Retention: $16,420.62

Reinsurer Liability: $214,255.35

Value at Risk (VaR) at 95.0%: $217.09

Conditional Tail Expectation (CTE) at 95.0%: $110.87

# At 85% we attain the below :

Surplus Share Retention: $16,420.62

Reinsurer Liability: $214,255.35

Value at Risk (VaR) at 85.0%: $792.54

Conditional Tail Expectation (CTE) at 85.0%: $376.31

# At 75%, the below is attained: 

Surplus Share Retention: $16,420.62

Reinsurer Liability: $214,255.35

Value at Risk (VaR) at 75.0%: $1,596.54

Conditional Tail Expectation (CTE) at 75.0%: $694.48

Given the above analysis, with reference to the confidence intervals of 0.95, 0.85 and 0.75, we are attaining

that at the 95% C.I the VaR,that our loss does not  exceed the amount of $ 217 which is best suite for our pricing model.

At the pricing model of 95%, we attain an optimal solution that necessitates that our liqiudity per portfolio is not affected, subject to the liabilities.